litellm/main.py

```diff
@@ -523,8 +523,13 @@ async def acompletion(
         await asyncio.sleep(mock_delay)
 
     try:
+        # Remove the acompletion flag before calling sync completion
+        # This ensures the sync wrapper properly sets up callbacks
+        sync_completion_kwargs = completion_kwargs.copy()
+        sync_completion_kwargs.pop("acompletion", None)
```

Comment:
hmm on 2nd thought this looks incorrect. we actually need the sync function to know that `acompletion=True` because the sync func returns a coroutine if acompletion=True